presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcPageSink.java

```diff
@@ -165,6 +165,11 @@ else if (javaType == Slice.class) {
             throw new PrestoException(JDBC_NON_TRANSIENT_ERROR, e);
         }
         catch (SQLException e) {
+            SQLException nextExc = e.getNextException();
```

Comment:
This is not the only place were we wrap `SQLException` as `PrestoException`. Do we need to update all places like that?

Reply:
I think we should. So far we've only seen a SQLException chain from Redshift and hence this change was limited to this.

If we do want to do this everywhere then I'll need to think of a better way than copy pasting this everywhere.

Reply:
If we want to do this everywhere, we'll probably need to copy/paste _something_. I think subclassing `PrestoException` (`JdbcPrestoException`) or adding factory methods somewhere are preferable to just adding a utility method that converts chained `SQLException`s to suppressed exceptions.

(The only way I can think of to avoid needing to make changes in a lot of places would be to add constructors accepting `SQLException` to `PrestoException`, but that seems like a _very_  bad idea.)